lib/spack/spack/cmd/flake8.py

```diff
@@ -5,200 +5,23 @@
-        "-r",
-        "--root-relative",
-        action="store_true",
-        default=False,
-        help="print root-relative paths (default: cwd-relative)",
-    )
-    subparser.add_argument(
-        "-U",
-        "--no-untracked",
-        dest="untracked",
-        action="store_false",
-        default=True,
-        help="exclude untracked files from checks",
-    )
-    subparser.add_argument(
-        "files", nargs=argparse.REMAINDER, help="specific files to check"
-    )
+    spack.cmd.style.setup_parser(subparser)
 
 
 def flake8(parser, args):
-    file_list = args.files
-    if file_list:
-
-        def prefix_relative(path):
-            return os.path.relpath(
-                os.path.abspath(os.path.realpath(path)), spack.paths.prefix
-            )
-
-        file_list = [prefix_relative(p) for p in file_list]
-
-    returncode = 0
-    with working_dir(spack.paths.prefix):
-        if not file_list:
-            file_list = changed_files(args.base, args.untracked, args.all)
-
-        print("=======================================================")
-        print("flake8: running flake8 code checks on spack.")
-        print()
-        print("Modified files:")
-        for filename in file_list:
-            print("  {0}".format(filename.strip()))
-        print("=======================================================")
-
-        output = ""
-        # run in chunks of 100 at a time to avoid line length limit
-        # filename parameter in config *does not work* for this reliably
-        for chunk in grouper(file_list, 100):
-            flake8_cmd = which("flake8", required=True)
-            chunk = filter(lambda e: e is not None, chunk)
-
-            output = flake8_cmd(
-                # use .flake8 implicitly to work around bug in flake8 upstream
-                # append-config is ignored if `--config` is explicitly listed
-                # see: https://gitlab.com/pycqa/flake8/-/issues/455
-                # "--config=.flake8",
-                *chunk,
-                fail_on_error=False,
-                output=str
-            )
-            returncode |= flake8_cmd.returncode
-
-            if args.root_relative:
-                # print results relative to repo root.
-                print(output)
-            else:
-                # print results relative to current working directory
-                def cwd_relative(path):
-                    return "{0}: [".format(
-                        os.path.relpath(
-                            os.path.join(spack.paths.prefix, path.group(1)),
-                            os.getcwd(),
-                        )
-                    )
-
-                for line in output.split("\n"):
-                    print(re.sub(r"^(.*): \[", cwd_relative, line))
-
-    if returncode != 0:
-        print("Flake8 found errors.")
-        sys.exit(1)
-    else:
-        print("Flake8 checks were clean.")
+    print("=======================================================")
```

Comment:
can this stuff be `tty.warn`?  Users will recognize that.  Suggest replacing everything with:

```python
tty.warn("`spack flake8` is deprecated", "please use `spack style` to run style checks.")
```